### PR TITLE
Refactor and cleanup byte utilities

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This refactor / cleanup is in preparation of an upcoming PR.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart [hunk 1]
```diff
@@ -318,7 +318,11 @@ class _GCHeapCapacityColumn extends _GCHeapStatsColumn {
 
   @override
   String getDisplayValue(AdaptedProfile dataObject) {
-    return prettyPrintBytes(getValue(dataObject), includeUnit: true)!;
+    return prettyPrintBytes(
+      getValue(dataObject),
+      includeUnit: true,
+      kbFractionDigits: 0,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart:324]: @polina-c should these columns actually have KB fraction digits of size 1? This PR changes the default value for this parameter so I added the 0 here to keep the current behavior, but wanted to check if this is the current behavior by accident or if it is intentional.

Comment (jacob314) [reply packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart:324]: agree that this should probably be changed to 1.

File: packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart [hunk 2]
```diff
@@ -421,6 +422,7 @@ class ImageResponseView extends StatelessWidget {
               prettyPrintBytes(
                 encodedResponse.lengthInBytes,
                 includeUnit: true,
+                kbFractionDigits: 0,
```
Comment (jacob314) [inline packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart:425]: doesn't really matter here as images are large but could probably be 1 for consistency.

File: packages/devtools_app/lib/src/shared/primitives/byte_utils.dart [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  int mbFractionDigits = 1,
+  int gbFractionDigits = 1,
+  bool includeUnit = false,
+  num roundingPoint = 1.0,
+  int maxBytes = 52,
+}) {
+  if (bytes == null) {
+    return null;
+  }
+  // TODO(peterdjlee): Generalize to handle different kbFractionDigits.
```
Comment (jacob314) [inline packages/devtools_app/lib/src/shared/primitives/byte_utils.dart:23]: nit: change TODO to your name while moving code as peter is unlikely to resolve the todo.

**Review by jacob314 — APPROVED** [review]
LGTM with a couple suggestions that are better for a followup than this pull request.

File: packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart:324]: done

**Review by jacob314 — APPROVED** [review]
lgtm

**Pull request merged**